gateway-service/src/main/java/org/zowe/apiml/gateway/security/service/saf/SafIdtProvider.java

```diff
@@ -9,25 +9,26 @@
  */
 package org.zowe.apiml.gateway.security.service.saf;
 
-import java.util.Optional;
-
 /**
- * It's possible to configure various SafIdtProviders. At the moment only one configured at the time is allowed. If there
- * are multiple providers configures, the behavior could be unpredictable.
+ * It's possible to configure various SafIdtProviders. At the moment only one configured at the time is allowed.
+ * If there are multiple providers configured, the behavior can be unpredictable.
  */
 public interface SafIdtProvider {
+
     /**
      * If the current user has the proper rights generate the SAF token on its behalf and return it back.
      *
      * @return Either empty answer meaning the user is either unauthenticated or doesn't have the proper rights.
      */
-    Optional<String> generate(String username);
+    String generate(String username, String applId);
```

Comment:
Why did you decide to go with Exceptions instead of Optional? 

The fact that some users doesn't have the proper rights isn't per see, something exceptional. Which is why the Optional was used in this place. 

Also why do you add the second method? It introduces more overhead to the implementations of this interface. Is there any strong reason to have somewhere just the implementation with the applId without password?

Reply:
I can remove the original implementation without a password if you prefer. The reason is that ZSS (and other implementations) can do both things during one call in the future - generate a passticket and generate IDT. This would allow gw to run off z.

Reply:
I believe it's better to remove the original one. The implementation can decide to ignore the password.

Reply:
There is one potential case. The original method could be used in the future to log in. The shorter one can reduce one more call to ZSS to obtain passticket, when API ML will be inside the Docker. This implementation is good for future cases and then it will be difficult to change interface I guess.

Reply:
So as for the use case where we call to the ZSS twice, once for the Passticket and once for the SAF IDT.

Unless we also change the API of ZSS to actually generate passticket on its own, there just will be two calls so this isn't removing one with the current APIs and implementations. 

And I believe that the ZSS API for generating SAF IDT should remain generic and therefore make it possible to use password instead of passticket where relevant.